Bug 1060915 - Accept view-source: protocol

takenspc opened the pull request:
This patch makes the browser app accept view-source: URLs. 

It also unify scheme checking logic because there were regexp patterns which cannot handle scheme containing "-".

Commit:
Bug 1102831 - Implement user dictionary UI to KeyboardSettingsApp

- Only stores the list of words to indexedDB and NOT the generation of dictionary blob, for now.
- Style KBSettingsApp to resemble Settings app more, pre UX request.
- KBSettingsApp now contains multiple panels & dialogs, and thus a PanelController has been added to control the transition sequence, with similar structure in Settings app.
- Add build integration test for handwriting and user dictionary settings

mancas committed:
Bug 1100341 - [Statusbar][Edge gestures] Icons overlay when performing edge gestures while app opening. r=etienne

Commit:
Bug 1106185 - Align answer/reject buttons on Callscreen, incl. RTL support. r=gtorodelvalle

Commit:
Bug 1106624 - [Downloads] Music file does not open/play from the download list

Commit:
Bug 1101230: Properly create camera's preview images.

mwargers committed:
Bug 1112334 - Write a test for basic Find My Device logging in existing account

jrburke committed:
Bug 1086598 - [Clock] Due to AMD, showing alarm screen takes ~7 sec

gmarty committed:
Bug 1093388 - [Browser] Requires an internet connection error is overlapped by check settings in landscape mode

Commit:
bug 1098598 - Use default crashedAppName in crash reporter. r=kgrandon

jrburke committed:
Bug 1042361 - [email] Avoid disruptive notification modes if a notification for the account already exists

azasypkin committed:
Bug 1112135 - [Messages] When starting as an activity/notification/, delay threads rendering until the right panel is open. r=julien

Commit:
Bug 1102712: honor devicePixelRatio in Gallery image editor

always round up or down when coverting between CSS and device pixels

Commit:
Bug 1114326 - [Stingray] Grid-view of app deck could not scroll down when app icons are out of screen

JohanLorenzo committed:
Bug 1113188 - Add a test to check deletion of multiple contacts

Commit:
Bug 990481 - [ZTE][OPENC]MMS app allows to attach more contents even when the max MMS size is reached r=zasypkin

julienw committed:
Bug 1112868 - [Messages] Message Options appear alongside Keyboard; blocks view of text entry r=schung,azasypkin

Blur the focus asap when displaying OptionMenu or Dialog.

Commit:
Bug 1101136 - Make sure that alarm messages don't get dropped within calendar

cctuan committed:
Bug 1114640 -  Create a new test case for the rating feature of the Music App

Commit:
Bug 1033334 - [Message] We should keep the focus in composer after sending a message r=Steve Chung

snowmantw committed:
Bug 1104351 - [System Lockscreen]The lockscreen slider will not reset when holding it in place and pressing the power button

cctuan committed:
Bug 1102495 - [Settings] Provide an option to enable copy/paste in the Developer menu

snowmantw committed:
Bug 1097994 - [Lockscreen][Camera] Keypad is displayed before Camera app
is launched from the Lockscreen with a passcode.

rexboy7 committed:
Bug 1105982 - [Stingray][home] Add new folder

Commit:
Bug 1115807 - Use <meta> keys in favor of manifest.webapp for localization metadata

Commit:
Bug 1115807 - Migrate all HTML files from manifest.webapp to <meta> l10n

Commit:
Bug 1115807 - Update l10n.js to use <meta> for l10n

Commit:
Bug 1116071 - Video headers in RTL have no enough space before them

ochameau committed:
Bug 1102972 - ensure rebuilding app removed from profile folder. r=rickychien

Commit:
Bug 1116406 - Run specific test-file when executing build unit/integration tests

EdgarChen committed:
Bug 1114955 - Apn doesn't be saved correctly into database during device first boot up

mwargers committed:
Bug 1116456 - [v2.2] Investigate test_marketplace_packaged_app.py failure

JohanLorenzo committed:
Bug 1114568 - Add a test to check that a regular phone number should not be dialed out on the emergency dialer

mwargers committed:
Bug 1086687 - Write unlink facebook friends test

Commit:
Bug 1114643 - Create a new test case for cancelling the deletion of multiple photos in gallery

asutherland committed:
Bug 1116087 - Safely clone elements for HTML cookie cache

Bug 1081039 fixed the platform to call the custom element callback
appropriately when cloning / importing nodes.  A document context
with a fresh registry is required to do this (actually) correctly.

Without this fix the email app's HTML cookie cache would end up
with a corrupted cookie cache state and potentially other app
problems since the rest of the card infrastructure really was not
expecting this type of thing to happen.

takenspc committed:
Bug 1060915 Accept view-source: protocol

Commit:
Bug 1109805: Deduplicate mutation targets before calling translate*. r=stas

tamarahills committed:
Bug 990369 - For STK Call control change number on connect.

Commit:
Bug 1115295 - [Stingray][tv_shared] Remove empty folders correctly when flushing

takenspc committed:
Bug 1069612 - Switch tabs button is not labeled

This adds labels to Back and Forward buttons too.

julienw committed:
Bug 1113002 - Remove unused function from utils.js r=azasypkin

Commit:
Bug 1116162 - Expose console.log|warn|info|error when loading l10n.js at buildtime.

KevinGrandon committed:
Bug 1116986 - Quick fix for FxA failing test by hard-coding additional years r=kgrandon

ferjm committed:
Bug 1116986 - (Follow-up) Dynamically populate year select. r=kgrandon

Commit:
Bug 1117002 - Deduplicate AST entries for ab-CD.json

Commit:
Bug 1117002 - Deduplicate AST entries for ab-CD.json

albertopq committed:
Bug 1114764 - Statusbar visibility on inline activities

Commit:
Bug 1115321 - Remove calendar event notification after user taps on it

JohanLorenzo committed:
Bug 1117115 - Enable test_settings_sim_manager.py

KevinGrandon committed:
Revert "Bug 1117002 - Deduplicate AST entries for ab-CD.json"

This reverts commit fee0d623346c34b7eaa860141158f2fc57913391.

takenspc committed:
Bug 1117473 - [E-Mail] Typos in manifest.webapp

KevinGrandon committed:
Bug 1106411 - [PrivacyPanel] Hide on the vertical home screen by default

snowmantw committed:
Bug 1090451 - [Lockscreen] No vibration when entering passcode

huchengtw-moz committed:
Bug 1113499 - [Stingray][Smart-home] Open app with specified hash/URL

julienw committed:
Bug 1109770 - [gaia-header] trigger the reformat only once even if several mutations happen, for the same heading r=gmarty

etiennesegonzac committed:
Bug 1114800 - Don't let the app element and the app iframe size get out of sync. r=alive

Commit:
Bug 1113134 - Second phone number gets duplicated if you delete the first one and press "add phone" r=ferjm +autolander

huchengtw-moz committed:
Bug 1113495 - [Stingray][Smart-system] Accept specified URL from home on opening app

KevinGrandon committed:
Bug 1109529 - [Collection] Allow apps of role=search to be added to collections r=crdlc

KevinGrandon committed:
Bug 1116986 - (Follow-up) Remove unneeded mozL10n.once. r=kgrandon

sfoster committed:
Bug 1093822 - Update notifications tutorial string. r=fcampo

Commit:
Bug 1118115 - Add build script test document for TEST_FILES

cctuan committed:
Bug 1110963 - [Text Selection] Cut/Copy/Paste menu does not dismiss when deleting selected text

gasolin committed:
Bug 1115622 - [Bluetooth] decouple bluetoothTransfer and NfcHandoverManager's dependency

Commit:
Bug 1117694 - Fix a regression in bluetooth handling caused by bug 1095109

KevinGrandon commented on shared/js/url_helper.js:
```diff
@@ -33,10 +31,10 @@ var UrlHelper = {
+    var case3Reg = /^(data|view-source)\:/;
     // for cases, only scheme but no domain provided
-    var case4Reg = /^\w+\:\/*$/;
+    var case4Reg = /^[a-z\u00a1-\uffff0-9-+]+\:\/*$/;
```
This looks like it's trying to match against the scheme. Instead of re-declaring a very similar regex here, can we use ```this.hasScheme()``` instead of the ```case4Reg.test()``` call in the next if statement?

dominickuo committed:
Bug 1103254 - [Music] correct the wrong path of the player view file

viorelaioia committed:
Bug 1117823 - Update locators in Settings - Cell data page

Commit:
Bug 1056008 - Remove tbpl-manifest.ini

KevinGrandon committed:
Bug 1118315 - [Calendar] Remove strings from index.html r=kgrandon

JohanLorenzo committed:
Revert "Bug 1117115 - Enable test_settings_sim_manager.py"

This reverts commit a2c375c0fe832807630b8d0b2174f790dad74897.

jrburke committed:
Bug 1113441 - [Email] Top appears on the top left of the screen when scrolling down email search results

Commit:
Bug 1113891 - Event remains on the notification screen after tapping on it

KevinGrandon committed:
Bug 1118498 - Jshint fixes for mock_activity.js r=kgrandon

snowmantw committed:
Bug 1104351 - [System Lockscreen]The lockscreen slider will not reset when holding it in place and pressing the power button (rev2)

timdream committed:
Bug 1102831 - Follow up, changes to CloseLockManager interface

takenspc committed:
Reduce number of regexps

- rscheme was changed a little as it previously ignored "//".
- UrlHelper._getScheme returns scheme + ":" or scheme + "://" or null.
